lib/spack/spack/build_systems/python.py

```diff
@@ -105,19 +106,38 @@ def build_directory(self):
         """The directory containing the ``setup.py`` file."""
         return self.stage.source_path
 
-    def python(self, *args):
-        inspect.getmodule(self).python(*args)
+    def python(self, *args, **kwargs):
+        inspect.getmodule(self).python(*args, **kwargs)
 
-    def setup_py(self, *args):
+    def setup_py(self, *args, **kwargs):
         setup = self.setup_file(self.spec, self.prefix)
 
         with working_dir(self.build_directory()):
-            self.python(setup, '--no-user-cfg', *args)
+            self.python(setup, '--no-user-cfg', *args, **kwargs)
+
+    def _setup_command_available(self, command):
+        """Determines whether or not a setup.py command exists.
+
+        :param str command: The command to look for
+        :return: True if the command is found, else False
+        :rtype: bool
+        """
+        kwargs = {
+            'output': os.devnull,
+            'error':  os.devnull,
+            'fail_on_error': False
+        }
+
+        python = inspect.getmodule(self).python
+        setup = self.setup_file(self.spec, self.prefix)
+
+        python(setup, '--no-user-cfg', command, '--help', **kwargs)
+        return python.returncode == 0
```

Comment:
This seems kind of hacky, but I couldn't think of a better way to tell whether or not `python setup.py test` is available. I also don't like how it prints the test line when we run with `spack install --verbose --run-tests`. If we could make that configurable, it would be nice.